lib/slax_web/websockets/disable.ex

```diff
@@ -0,0 +1,199 @@
+defmodule SlaxWeb.Disable do
+  @moduledoc """
+  A module that handles Slack websocket payloads for disabling Slax interactions and
+  builds modal views for slack https://api.slack.com/reference/surfaces/views
+  """
+
+  alias Slax.Slack
+  alias Slax.Channels
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "shortcut",
+        "callback_id" => "disable_slax"
+      }) do
+    view = build_disable_view(%{trigger_id: trigger_id})
+
+    Slack.open_modal(%{trigger_id: trigger_id, view: view})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "shortcut",
+        "callback_id" => "enable_slax"
+      }) do
+    view = build_enable_view()
+
+    Slack.open_modal(%{trigger_id: trigger_id, view: view})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => _trigger_id,
+        "type" => "view_submission",
+        "view" => %{
+          "callback_id" => "disable_view",
+          "state" => %{
+            "values" => values
+          }
+        }
+      }) do
```

Comment:
These function heads are doing a thing that, when taken to far, can make code a lot harder to understand. It uses pattern matching for _two different purposes_: to extract values for use later on AND to control which function clause is invoked based on the input 

The former use case (extracting things for use later) is purely a convenience, so generally I'd prefer to pull that out of the function head and into the body

https://hexdocs.pm/elixir/1.16.0-rc.0/code-anti-patterns.html#complex-extractions-in-clauses

Reply:
This is interesting. Thanks for pointing it out